model/textparse/openmetricsparse_test.go

```diff
@@ -486,7 +486,9 @@ func TestUTF8OpenMetricsParse(t *testing.T) {
 {"http.status",q="0.9",a="b"} 8.3835e-05
 {q="0.9","http.status",a="b"} 8.3835e-05
 {"go.gc_duration_seconds_sum"} 0.004304266
-{"Heizölrückstoßabdämpfung 10€ metric with \"interesting\" {character\nchoices}","strange©™\n'quoted' \"name\""="6"} 10.0`
+{"Heizölrückstoßabdämpfung 10€ metric with \"interesting\" {character\nchoices}","strange©™\n'quoted' \"name\""="6"} 10.0
```

Comment:
Note if the closing quote here was on the next line, we wouldn't get a diff on this line.
Suggest you break the cycle by changing your line 491 to do this.

Reply:
Unfortunately that actually breaks on parsing (`expected a valid start token, got "\n" ("INVALID") while parsing: "\n"`), so it'd also require a trim of the string before parsing. I can include that, but the extra trim might look a bit odd to future reviewers.

Reply:
yes, having a newline before the closing quote causes the parser to attempt to parse another line and hit EOF straight away, which raises the above error.